common/module.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/amp-labs/connectors/common/handy"
+)
+
+// ErrInvalidModuleDeclaration occurs when the identifier used for map indexing in Modules
+// does not match the module's ID.
+var ErrInvalidModuleDeclaration = errors.New("supported modules are not correctly defined")
+
+type ModuleID string
+
+type Modules = handy.Map[ModuleID, Module]
+
+// Module represents a set of endpoints and functionality available by provider.
+// Single provider may support multiple modules, requiring the user to choose a module before making requests.
+// Modules may differ by version or by theme, covering different systems or functionalities.
+type Module struct {
```

Comment:
Module was moved out of `paramsbuilder` to `common` package.
This eliminates name collisions, and this concept doesn't really belong to building parameters. 
The parameter will own it. Module is then propagated and saved under connector struct and is later used for it's `ID` and `Path()`. Less separate data to store as you requested.

@RajatPawar